tqecjs/src/workspace/plaquettes/PlaquetteClass.js

```diff
@@ -0,0 +1,370 @@
+import { Graphics, Container } from 'pixi.js';
+import { button } from '../components/button';
+
+export default class Plaquette extends Graphics {
+	constructor(qubits, workspace, gridSize = 50, color = 'purple') {
+		super();
+		// UI properties
+		this.workspace = workspace;
+		this.color = color;
+		this.gridSize = gridSize;
+		this.gridOffsetX = this.gridSize;
+		this.gridOffsetY = this.gridSize;
+		this.isDragging = false;
+		this.plaquetteMade = false;
+		this.showButton = false;
+		this.controlPanel = new Container();
+		this.controlPanel.name = 'control_panel';
+		this.controlPanel.visible = false;
+		this.rotateButton = button('Rotate', 200, 200, 0x000000);
+		this.clearButton = button('Clear', 200, 275, 'red');
+		this.colorButton = button('Change color', 200, 350, 'green');
+		this.makeRotatable(); // Add the rotate button to screen
+		this.clearPlaquette(); // Add the clearbutton to the screen
+		this.changeColorButton(); // Add the change color button to the screen
+		//this.changeColorButton(); // Addh the color change button
+		// QC properties
+		this.qubits = qubits || []; // We assume that the list is the order of the qubits
+		this.qubitStack = []; // The stack of qubits that form the convex hull
+		this.createPlaquette(); // Create the plaquette
+		this.quantumCircuit = null; // Might be better to have a quantum circuit class
```

Comment:
Would it make sense to use something like Q.js?

https://quantumjavascript.app/Q-Circuit.html